eng/pipelines/templates/Apex_Tests_On_Windows.yml

```diff
@@ -23,7 +23,7 @@ steps:
     solution: "build\\bootstrap.proj"
     msbuildVersion: "16.0"
     configuration: "$(BuildConfiguration)"
-    msbuildArguments: "/t:Restore"
+    msbuildArguments: "/t:Restore -bl:$(Build.Repository.LocalPath)\\artifacts\\binlog\\bootstrap.binlog"
```

Comment:
I'd suspect this adds some performance overhead doesn't it?

You think the benefit outweighs the cost?
How often would a binlog have helped the investigations?

Reply:
I'm not trying to make an argument to not take this change, I'm mostly curious :)

Reply:
indeed. I was thinking how it could be off by default and turned on via a build variable at queue time, but since every step needs a unique argument, I couldn't think of a good way to do it.

Any ideas?

Reply:
Maybe? :) 

Early in the pipeline, add a conditional task that given a value would set another `N` variables, which are effective `-bl:binlogFile.Name`. 

Most of the time those variables would be `` so nothing will happen, but when the special variable is set, you get a binlog!